src/ImageSharp/Metadata/Profiles/Exif/ExifReader.cs

```diff
@@ -5,28 +5,45 @@
 using System.Buffers.Binary;
 using System.Collections.Generic;
 using System.Collections.ObjectModel;
+using System.Diagnostics;
+using System.IO;
 using System.Runtime.CompilerServices;
 using System.Text;
 
 namespace SixLabors.ImageSharp.Metadata.Profiles.Exif
 {
     /// <summary>
-    /// Reads and parses EXIF data from a byte array.
+    /// Reads and parses EXIF data from a stream.
     /// </summary>
-    internal sealed class ExifReader
+    internal class ExifReader
     {
-        private List<ExifTag> invalidTags;
-        private readonly byte[] exifData;
-        private int position;
+        private readonly Stream data;
+
+        private readonly byte[] offsetBuffer = new byte[4];
+        private readonly byte[] buf4 = new byte[4];
+        private readonly byte[] buf2 = new byte[2];
+
+        // used for sequential read big values (actual for multiframe big files)
+        // todo: different tags can link to the same data (stream offset) - investigate
+        private readonly SortedList<uint, Action> lazyLoaders = new SortedList<uint, Action>(new DuplicateKeyComparer<uint>());
+
         private bool isBigEndian;
-        private uint exifOffset;
-        private uint gpsOffset;
 
-        public ExifReader(byte[] exifData)
+        private List<ExifTag> invalidTags;
+
+        private uint exifOffset = 0;
+
+        private uint gpsOffset = 0;
+
+        public ExifReader(bool isBigEndian, Stream stream)
         {
-            this.exifData = exifData ?? throw new ArgumentNullException(nameof(exifData));
+            this.isBigEndian = isBigEndian;
+            this.data = stream ?? throw new ArgumentNullException(nameof(stream));
         }
 
+        public ExifReader(byte[] exifData) =>
+            this.data = new MemoryStream(exifData ?? throw new ArgumentNullException(nameof(exifData)));
```

Comment:
endianness will readed later: https://github.com/SixLabors/ImageSharp/blob/9f54406aa83002c9e270af937c0b92c1d76d6f83/src/ImageSharp/Metadata/Profiles/Exif/ExifReader.cs#L74

the class turned out to be a little universal, it can make 2 inheritor for tiff and exif, what do you think?

Reply:
@JimBobSquarePants it seems, worth to make 2 inheritor for tiff and exif, it will look better

Reply:
@JimBobSquarePants please review again, I split the classes and they seems to be looking better now